packages/jaeger-ui/src/components/App/TopNav.test.js

```diff
@@ -73,17 +73,17 @@ describe('<TopNav>', () => {
   describe('renders the default menu options', () => {
     it('renders the "JAEGER UI" link', () => {
       const items = wrapper.find(Link).findWhere(link => /JAEGER UI/.test(link.text()));
-      expect(items.length).toBe(1);
+      expect(items.length).toBe(2);
```

Comment:
What is the meaning of this change in the test? Are there two JAEGER UI buttons now?

Reply:
No, since we have wrapped the JaegerUIApp into an browser router all of the Nav links are being counted twice as one of it is under the parent browserRouter. Should I add an commented line to explain this ambiguity?

Reply:
I am not sure I follow, there still should be only one link regardless of how it is wrapped above.

Reply:
Hi @yurishkuro , so I debugged this further and it looks like enzyme is detecting two Jaeger-UI, one from the react component and other from the rendered Dom code https://github.com/enzymejs/enzyme/issues/836#issuecomment-401260477 . There is also an issue over here https://github.com/enzymejs/enzyme/issues/836 which solves the problem by using hostNodes but to use hostNodes, we would need to change the test selector.

Reply:
I even printed out the debug wrapper from enzyme shallow and it contained the correct code
```js
      <div>
        <Menu theme="dark" mode="horizontal" selectable={false} className="ub-right" selectedKeys={{...}} prefixCls="ant-menu" focusable={false}>
          <MenuItem>
            <withRouter(TraceIDSearchInput) />
          </MenuItem>
          <MenuItem disabled={false}>
            <a href="https://github.com/uber/jaeger" target="_self" rel="noopener noreferrer">
              GitHub
            </a>
          </MenuItem>
          <MenuItem disabled={false}>
            <a href="https://medium.com/jaegertracing/" target="_blank" rel="noopener noreferrer">
              Blog
            </a>
          </MenuItem>
          <MenuItem>
            <CustomNavDropdown label="About Jaeger" items={{...}} />
          </MenuItem>
        </Menu>
        <Menu theme="dark" mode="horizontal" selectable={false} selectedKeys={{...}} prefixCls="ant-menu" className="" focusable={false}>
          <MenuItem>
            <Link to="/" style={{...}}>
              JAEGER UI
            </Link>
          </MenuItem>
          <MenuItem>
            <Link to="/search">
              Search
            </Link>
          </MenuItem>
          <MenuItem>
            <Link to="/trace/...">
              Compare
            </Link>
          </MenuItem>
          <MenuItem>
            <Link to="/dependencies">
              System Architecture
            </Link>
          </MenuItem>
        </Menu>
      </div>
```

however I am not sure why is this error coming after the upgrade to react-router-dom v5.2 😕

Reply:
Any help here @rubenvp8510 @yurishkuro 😕?

Reply:
This goes above my head a bit since I am not familiar with the frameworks, but from the enzyme issue you linked it sounds like you just need to adjust the selection to only return the type of node that's relevant, and then have the assertion only on size=1

Reply:
Hi @yurishkuro , I have done the changes please do take a look !